src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1901,7 +1901,6 @@ NOTE: if you're sure you want to do this, please open an issue as to why. In the
 
         let mut targetflags = flags;
         targetflags.push(format!("-Lnative={}", builder.test_helpers_out(target).display()));
-        targetflags.extend(linker_flags(builder, compiler.host, LldThreads::No));
```

Comment:
Shouldn't we design `linker_flags` in a way that preserves the old logic for tests instead of removing it entirely?

Reply:
I saw only very minimal gains with LLD vs the default Linux linker on UI tests, but it's true that on some other platforms the difference might be larger.

The problem with the old approach is that it cannot respect the "self-contained" more. So when you use `lld = "self-contained"`, the old approach ignores that and just uses the global `lld` for the tests.

I guess that a compromise could be to use the linker flag with `use-lld = true` and the new compiler flag with `use-lld = "self-contained"`, but I would eventually like to set `use-lld = "self-contained"` to be the default on CI; with this approach, some tests would simply fail..

Reply:
> I guess that a compromise could be to use the linker flag with use-lld = true and the new compiler flag with use-lld = "self-contained", but I would eventually like to set use-lld = "self-contained" to be the default on CI; with this approach, some tests would simply fail..

This sounds reasonable I think.

Reply:
You said "eventually", I am not sure why we should concern on that part right now.

Reply:
It's not just about CI. Right now, `use-lld` is quite useful to make local compiler builds faster. But if we actually make `self-contained` working as intended, and keep the compiletest target flags, then `use-lld = "self-contained"` will also cause UI tests to fail for people that use this config value.

Reply:
I see. In this case I am fine with landing it as is. We can create an issue to track that part.